tests/integration/security/jwt_test.go

```diff
@@ -52,292 +49,278 @@ func TestAuthnJwt(t *testing.T) {
+							Headers: map[string][]string{
+								authHeaderKey: {"Bearer " + testIssuer1Token},
 							},
-							ExpectAuthenticated: false,
 						},
-						{
-							Request: connection.Checker{
-								From: a,
-								Options: echo.CallOptions{
-									Target:   b,
-									PortName: "http",
-									Scheme:   scheme.HTTP,
-									Path:     "/testing-istio-jwt",
-									Headers: map[string][]string{
-										authHeaderKey: {"Bearer " + testIssuer1Token},
-									},
-								},
+					},
+					ExpectAuthenticated: true,
+				},
+				{
+					Name: "jwt-two-issuers-token2",
+					Request: connection.Checker{
+						From: a,
+						Options: echo.CallOptions{
+							Target:   e,
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Path:     "/testing-istio-jwt",
+							Headers: map[string][]string{
+								authHeaderKey: {"Bearer " + testIssuer2Token},
 							},
-							ExpectAuthenticated: true,
 						},
-						{
-							Request: connection.Checker{
-								From: a,
-								Options: echo.CallOptions{
-									Target:   b,
-									PortName: "http",
-									Scheme:   scheme.HTTP,
-									Path:     "/testing-istio-jwt",
-									Headers: map[string][]string{
-										authHeaderKey: {"Bearer " + testIssuer2Token},
-									},
-								},
+					},
+					ExpectAuthenticated: false,
+				},
+				{
+					Name: "jwt-wrong-issuers",
+					Request: connection.Checker{
+						From: a,
+						Options: echo.CallOptions{
+							Target:   e,
+							Path:     "/wrong_issuer",
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Headers: map[string][]string{
+								authHeaderKey: {"Bearer " + testIssuer1Token},
 							},
-							ExpectAuthenticated: false,
 						},
 					},
+					ExpectAuthenticated: false,
 				},
 			}
 
 			for _, c := range testCases {
-				testName := strings.TrimSuffix(c.configFile, filepath.Ext(c.configFile))
-				t.Run(testName, func(t *testing.T) {
-
-					// Apply the policy.
-					namespaceTmpl := map[string]string{
-						"Namespace": ns.Name(),
-					}
-					deploymentYAML := tmpl.EvaluateAllOrFail(t, namespaceTmpl,
-						file.AsStringOrFail(t, filepath.Join("testdata", c.configFile)))
-					g.ApplyConfigOrFail(t, ns, deploymentYAML...)
-					defer g.DeleteConfigOrFail(t, ns, deploymentYAML...)
-
-					// Give some time for the policy propagate.
-					time.Sleep(60 * time.Second)
-					for _, subTest := range c.subTests {
-						subTestName := fmt.Sprintf("%s->%s:%s",
-							subTest.Request.From.Config().Service,
-							subTest.Request.Options.Target.Config().Service,
-							subTest.Request.Options.PortName)
-						t.Run(subTestName, func(t *testing.T) {
-							retry.UntilSuccessOrFail(t, subTest.CheckAuthn, retry.Delay(time.Second), retry.Timeout(10*time.Second))
-						})
-					}
+				t.Run(c.String(), func(t *testing.T) {
+					retry.UntilSuccessOrFail(t, c.CheckAuthn,
+						retry.Delay(250*time.Millisecond), retry.Timeout(30*time.Second))
```

Comment:
Is this possible to get false positive? I.e test pass because policy is not propagated. Not saying we don't have that issue before, but may be smt need to be watched out.

Reply:
This should be okay, the test cases expecting `deny` provides the most value here. A test case expecting allow is likely to always pass whether the policy is propagated or not.